clusterman/kubernetes/kubernetes_cluster_connector.py

```diff
@@ -215,14 +182,16 @@ def _is_node_safe_to_kill(self, node_ip: str) -> bool:
         return True
 
     def _get_nodes_by_ip(self) -> Mapping[str, KubernetesNode]:
-        pool_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
-        pool_nodes = self._core_api.list_node().items
+        use_different_label_for_nodes = self.pool_config.read_bool("use_different_label_for_nodes", default=False)
+
+        if use_different_label_for_nodes:
+            node_label_selector = self.pool_config.read_string("node_label_key", default="clusterman.com/pool")
+        else:
+            node_label_selector = self.pool_config.read_string("pool_label_key", default="clusterman.com/pool")
```

Comment:
ah, looking at this more closely I think our decision to use a toggle might be more effort than automatically trying to use the "right" one

e.g., 
```suggestion
        # TODO(CLUSTERMAN-XXX): we'll want to switch to using just pool_label_key once the new label is applied everywhere
        node_label_selector = self.pool_config.read_string(
        "node_label_key", 
        default=self.pool_config.read_string(
            "pool_label_key",
            default="clusterman.com/pool"
            )
        )     
```

where we try to use the temporary `node_label_key` if present in srvconfigs and otherwise we just fall back to what will eventually be the "permanent" key

that said, if we're planning on having all the nodes switched to the new label relatively quickly (like, by the end of the month or something) then it probably doesn't matter too much what we do here :p